clientv3/retry.go

```diff
@@ -27,12 +27,30 @@ type retryRpcFunc func(context.Context, rpcFunc) error
 
 func (c *Client) newRetryWrapper() retryRpcFunc {
 	return func(rpcCtx context.Context, f rpcFunc) error {
+		nrAuthRetries := 0
+
 		for {
 			err := f(rpcCtx)
 			if err == nil {
 				return nil
 			}
 
+			if grpc.Code(err) == codes.Unauthenticated {
+				// handle a case of expired auth token
+				if 0 < nrAuthRetries {
+					// second time, auth information e.g. user or role can be changed
+					return err
+				}
+				token, gterr := c.getToken()
+				if gterr != nil {
+					return err // return the original error for simplicity
+				}
+
+				c.tokenCred.token = token
```

Comment:
data race? maybe do this assignment in getToken() instead of returning the token and assigning here?